### ISSUE TITLE
typesafe config getting serialized caused "java.io.UTFDataFormatException: encoded string too long"

### ISSUE THREAD

**Issue by mengxr** [JohnSnowLabs/spark-nlp#772]
## Description

I saw [ConfigLoader.retrieve](https://github.com/JohnSnowLabs/spark-nlp/blob/master/src/main/scala/com/johnsnowlabs/util/ConfigLoader.scala#L21) is referenced in class objects that go into a serialized Spark task. This would lead to the "java.io.UTFDataFormatException" issue on some platform when the default (fallback) config is too big.

#745 #691 #679 might have the same root cause.

cc: @maziyarpanahi 

## Expected Behavior

No serialization error.

## Current Behavior

Serialization error on some platforms with large default typesafe config.

## Possible Solution

* Do we need to serialize the conf with spark tasks or does it get pulled into the task closure by accident? If it is not needed, we can mark it as `@transient`.
* If it is needed, I think the solution is to map the typesafe config to a narrow scoped config object that contains only relevant configs to avoid serialization issue.

## Steps to Reproduce

~~~scala
import com.johnsnowlabs.util.ConfigLoader
import java.io._

val bos = new ByteArrayOutputStream()
val oos = new ObjectOutputStream(bos)
oos.writeObject(ConfigLoader.retrieve)
oos.close()
~~~

throws the following error on Databricks Runtime 6.3 ML:

~~~
java.io.UTFDataFormatException: encoded string too long: 67385 bytes
	at java.io.DataOutputStream.writeUTF(DataOutputStream.java:364)
	at java.io.DataOutputStream.writeUTF(DataOutputStream.java:323)
	at com.typesafe.config.impl.SerializedConfigValue.writeValueData(SerializedConfigValue.java:301)
	at com.typesafe.config.impl.SerializedConfigValue.writeValue(SerializedConfigValue.java:375)
	at com.typesafe.config.impl.SerializedConfigValue.writeValueData(SerializedConfigValue.java:315)
	at com.typesafe.config.impl.SerializedConfigValue.writeValue(SerializedConfigValue.java:375)
	at com.typesafe.config.impl.SerializedConfigValue.writeValueData(SerializedConfigValue.java:315)
	at com.typesafe.config.impl.SerializedConfigValue.writeValue(SerializedConfigValue.java:375)
	at com.typesafe.config.impl.SerializedConfigValue.writeValueData(SerializedConfigValue.java:315)
	at com.typesafe.config.impl.SerializedConfigValue.writeValue(SerializedConfigValue.java:375)
	at com.typesafe.config.impl.SerializedConfigValue.writeExternal(SerializedConfigValue.java:441)
	at java.io.ObjectOutputStream.writeExternalData(ObjectOutputStream.java:1459)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1430)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1178)
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:348)
~~~

Unfortunately, we have a very large conf:

~~~scala
import com.typesafe.config.ConfigFactory
val config = ConfigFactory.load()
config.getConfig("java").getConfig("class").getString("path").length
~~~

output

~~~
67383
~~~

## Your Environment

* Spark NLP version: 2.3.4
* Spark version: 2.4.4 (Databricks Runtime 6.3 ML)
* Apache NLP version:
* Setup and installation (Pypi, Conda, Maven, etc.):
* Operating System and version:
* Link to your project (if any):

**Comment by maziyarpanahi** [comment]
@mengxr Thank you so much for debugging this and bringing it to our attention. In case of a large config provided by the user, this could happen and we will investigate this. However, those issues with the YARN cluster-mode I have tested and it works on some annotators and fails with the others. So clearly we have a feature of some sort that is not serializable when the Driver is part of the cluster itself.

This also could result in the same serialization issue which I am glad you brought it up so we can fix them all together 👍

**Comment by pmaditya** [comment]
Our team is also facing the same problem. Our Spark configuration is also greater than 75000 bytes and we are also seeing the same issue. 

Is there any ETA of a fix for this problem?

**Comment by maziyarpanahi** [comment]
We have made some progress and made some tests in Cloudera YARN cluster-mode and Databricks. The results are here: https://github.com/JohnSnowLabs/spark-nlp/issues/832

Could you please make a fat JAR from this PR and give it a go:
https://github.com/JohnSnowLabs/spark-nlp/pull/834

**Comment by maziyarpanahi** [comment]
This has been resolved in 2.4.5 release: https://github.com/JohnSnowLabs/spark-nlp/releases/tag/2.4.5
